src/main/java/com/patchworkmc/mapping/TargetMappingsHolder.java

```diff
@@ -0,0 +1,91 @@
+package com.patchworkmc.mapping;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import net.fabricmc.tinyremapper.IMappingProvider;
+
+import net.patchworkmc.manifest.accesstransformer.v2.exception.MissingMappingException;
+
+import com.patchworkmc.manifest.converter.accesstransformer.AccessTransformerConverter;
+
+/**
+ * A lazily loading representation of all Minecraft names and descriptors in the target mappings.
+ * Used in {@link AccessTransformerConverter} to resolve descriptors for fields and resolve all names for wildcards.
+ */
+public class TargetMappingsHolder {
+	private final HashMap<String, Class> mappings = new HashMap<>();
+
+	private final IMappingProvider targetFirst;
+
+	private boolean loaded = false;
+
+	/**
+	 * @param targetFirst the mappings, in the format {@code target -> <any>}.
+	 */
+	public TargetMappingsHolder(IMappingProvider targetFirst) {
+		this.targetFirst = targetFirst;
+	}
+
+	public Class getMappings(String owner) throws MissingMappingException {
+		// We load lazily here so that if this class isn't needed we can save some time and memory
+		if (!this.loaded) {
+			targetFirst.load(new Acceptor());
+			loaded = true;
+		}
+
+		Class result = mappings.get(owner);
+		MappingAssertions.assertClassExists(result, owner);
+		return result;
+	}
+
+	/**
+	 * POJO for {@link Member}s.
+	 */
+	public static class Class {
```

Comment:
To avoid confusion with the `Class` from Java.
```suggestion
	public static class ClassMembers {
```